src/mrpro/data/_KTrajectory.py

```diff
@@ -41,11 +43,99 @@ class KTrajectory:
 
     @property
     def broadcasted_shape(self) -> tuple[int, ...]:
-        """The broadcasted shape of the trajectory."""
+        """The broadcasted shape of the trajectory.
+
+        Returns
+        -------
+            broadcasted shape of trajectory
+        """
         shape = np.broadcast_shapes(self.kx.shape, self.ky.shape, self.kz.shape)
         return tuple(shape)
 
-    def as_tensor(self, stack_dim=0):
+    def _type_of_traj(
+        self, tolerance: float = 1e-2, orientation: Literal['kz-ky-kx', 'k2-k1-k0'] = 'kz-ky-kx'
+    ) -> list[TrajType]:
+        """Type of trajectory.
+
+        This function checks if the entries of the trajectory along certain dimensions
+            - are of shape 1 -> TrajType.SINGLEVALUE
+            - lie on a Cartesian grid -> TrajType.ONGRID
+            - none of the above -> TrajType.NOTONGRID
+
+        Parameters
+        ----------
+        tolerance
+            tolerance used to determine if trajectory points lie on a gird
+
+        Returns
+        -------
+            type of trajectory along 'kz-ky-kx' or 'k2-k1-k0'
+        """
+        # Matrix describing trajectory-type [(kz, ky, kx), (k2, k1, k0)]
+        # Start with everything not on a grid (arbitrary k-space locations).
+        # We use the value of the enum-type to make it easier to do array operations.
+        traj_type_matrix = torch.tensor([[TrajType.NOTONGRID.value] * 3] * 3, dtype=torch.int8)
+
+        for ind, ks in enumerate((self.kz, self.ky, self.kx)):
+            are_values_on_grid = (ks - ks.to(dtype=torch.int64).to(dtype=ks.dtype)) <= tolerance
```

Comment:
`torch.rint` ?

Reply:
I don't think torch.rint exists but torch.round does the job